contrib/devtools/lint-include-guards.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Check include guards.
+
+HEADER_ID_PREFIX="BITCOIN_"
+HEADER_ID_SUFFIX="_H"
+
+REGEXP_EXCLUDE_FILES_WITH_PREFIX="src/(crypto/ctaes/|leveldb/|secp256k1/|tinyformat.h|univalue/)"
+
+EXIT_CODE=0
+for HEADER_FILE in $(git ls-files -- "*.h" | grep -vE "^${REGEXP_EXCLUDE_FILES_WITH_PREFIX}")
```

Comment:
Could use `git ls-files -x` instead of `| grep -vE`?

Reply:
Nevermind, `-x` skips untracked files.

Reply:
@promag No I don't think so (`-x` is for untracked files)? :-)

I'd be glad to be proven wrong so provide an equivalent with `git ls-files -x` and I'll switch :-)

Reply:
Yes I've come to that conclusion right trying it.